packages/vscode-extension/src/devices/DeviceBase.ts

```diff
@@ -52,17 +52,33 @@ export abstract class DeviceBase implements Disposable {
     return this.preview?.hideTouches();
   }
 
-  public stopReplays() {
-    return this.preview?.stopReplays();
+  public startRecording() {
+    if (!this.preview) {
+      throw new Error("Preview not started");
+    }
+    return this.preview.startRecording();
+  }
+
+  public async captureAndStopRecording() {
+    if (!this.preview) {
+      throw new Error("Preview not started");
+    }
+    const recordingDataPromise = this.preview.captureRecording();
+    this.preview?.stopRecording();
+    return recordingDataPromise;
```

Comment:
for symmetry, it'd be better if this method just proxied the call down to preview instead of making two calls